sdk/program/src/borsh.rs

```diff
@@ -54,3 +60,10 @@ pub fn get_packed_len<S: BorshSchema>() -> usize {
     let schema_container = S::schema_container();
     get_declaration_packed_len(&schema_container.declaration, &schema_container.definitions)
 }
+
+/// Deserializes without checking that the entire slice has been consumed
+pub fn try_from_slice_unchecked<T: BorshDeserialize>(data: &[u8]) -> Result<T, Error> {
```

Comment:
How about a little test for this guy?

Reply:
Certainly

Reply:
Unchecked sounds potentially dangerous, should folks be cautious using this function and if so what should they pay attention to?

Reply:
Hm, that's a good point.  The main difference is that this function eliminates a check that the slice has been totally read, so it lets you work with overallocated buffers.  The normal `try_from_slice` returns an error if there's data left in the buffer after deserialization.  How could we clarify that?

Reply:
Is there a downside to always using the "unchecked" version?

Reply:
The length check ensures that whatever you're expecting to deserialize uses up all of the bytes in the buffer, so you could potentially run into an issue if you always use unchecked.  Any buffer that's big enough could work to deserialize your type, so if you pass the wrong buffer with unchecked, the error wouldn't get caught.

Reply:
Any buffer greater than or equal to the expected size would work, correct?  Maybe call it "unsized", "unchecked" is fine but should probably include something to effect of the description you have above as what the "unchecked" means practically to the developer

Reply:
Correct -- I'll rename the function and add the description then

Reply:
With the good description you have provided I think "unchecked" makes sense, I can go either way but "unchecked" is more consistent with rust norms

Reply:
It's also consistent with the PR that I put in for Borsh JS :-D